### PR TITLE
HJ-319 Add DBCache model and table

### PR DESCRIPTION
**Pull request by erosselli**
Implements part of [HJ-319](https://ethyca.atlassian.net/browse/HJ-319)

### Description Of Changes

Adds a `DBCache` model and table to be used for persistent caching. The table has the `namespace` and `cache_key` columns that uniquely determine a row, and the column `cache_value` is used for storing an arbitrary value. For now we'll only use this for the list privacy experiences endpoint, but in the future it should be easy to add other things as well. 

### Code Changes

- Create `DBCache` model and its corresponding migration 
  -  We index both over namespace (since we may want to implement a "clear all entries for a namespace") and over the (namespace, cache_key) pair (for lookups) 

### Steps to Confirm

1.  _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] Issue requirements met
* [ ] All CI pipelines succeeded
* [ ] `CHANGELOG.md` updated
* Followup issues:
  * [ ] Followup issues created (include link)
  * [ ] No followup issues
* Database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!
  * [ ] No migrations
* Documentation:
  * [ ] Documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] Documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] If there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
  * [ ] No documentation updates required

[HJ-319]: https://ethyca.atlassian.net/browse/HJ-319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/fides/api/models/db_cache.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    """
+    Cache table for storing arbitrary data, useful when in-memory caches aren't enough.
+    For example if cache contents need to be persisted across server restarts,
+    or if the cache needs to be shared across different Fides instances.
+    """
+
+    namespace = Column(
+        String, nullable=False, index=True
+    )  # Add a namespace since the same cache key could technically be used for different contexts
+    cache_key = Column(String, nullable=False, index=True)
```
Comment (adamsachs) [inline src/fides/api/models/db_cache.py:27]: hmm - i'm not sure how much it matters, but i _think_ for a composite index (which this is, right?) it's generally [specified a bit differently](https://docs.sqlalchemy.org/en/20/core/constraints.html#index-api)?

and we have some examples in our code base, e.g. [here](https://github.com/ethyca/fides/blob/main/src/fides/api/models/system_history.py#L30-L34)

File: src/fides/api/models/db_cache.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    cache_value = Column(BYTEA, nullable=False)
```
Comment (adamsachs) [inline src/fides/api/models/db_cache.py:28]: ok, i hope this works well and i wasn't silly to suggest a byte column :)

**Review by adamsachs — APPROVED** [review]
this looks solid to me! just a nit on the sqlalchemy representation of the composite index 👍

File: src/fides/api/models/db_cache.py (hunk 1 above)
Comment (erosselli) [reply src/fides/api/models/db_cache.py:27]: ah I forgot to specify the composite index on the model definition, I just added it in the migration 🤦‍♀️  I did add index=True just to namespace thinking of queries like "Let's remove all entries for a namespace" , that don't have a specific cache_key but maybe the composite index still works for those kinds of queries? I'm a bit rusty on the specifics of composite indexes 😅

Comment (adamsachs) [reply src/fides/api/models/db_cache.py:27]: heh, i'm not really an expert either. from my general understanding, i don't _think_ the composite index would work well specifically for that use case - but i haven't validated that assumption at all.

i was aware of your intention with the two indexes there, and i didn't really question it - so i'd say it's fine to keep that in place! 👍

**Pull request merged**